test/onnx/CMakeLists.txt

```diff
@@ -24,14 +24,19 @@
 
 
 function(add_onnx_test TEST_NAME)
-    add_executable(${TEST_NAME} ${ARGN})
+    rocm_add_test_executable(${TEST_NAME} ${ARGN})
     rocm_clang_tidy_check(${TEST_NAME})
-    target_link_libraries(${TEST_NAME} migraphx_onnx migraphx_ref)
+    target_link_libraries(${TEST_NAME} migraphx_onnx migraphx_ref onnx_files weight_files)
     target_include_directories(${TEST_NAME} PUBLIC ../include include)
-    add_test(NAME ${TEST_NAME} COMMAND $<TARGET_FILE:${TEST_NAME}> WORKING_DIRECTORY ${TEST_ONNX_DIR})
-    rocm_mark_as_test(${TEST_NAME})
 endfunction()
 
+include(Embed)
+file(GLOB_RECURSE ONNX_FILES CONFIGURE_DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/*.onnx)
+add_embed_library(onnx_files ${ONNX_FILES} RELATIVE ${CMAKE_CURRENT_SOURCE_DIR})
+
+file(GLOB_RECURSE WEIGHT_FILES CONFIGURE_DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/*.weight)
+add_embed_library(weight_files ${WEIGHT_FILES} RELATIVE ${CMAKE_CURRENT_SOURCE_DIR})
```

Comment:
There is no need to use two different embed libraries. We can distinguish it based on the name. Although, if you pick option 2, this may not even be necessary.

Reply:
Done